homeassistant/components/solaredge/sensor.py

```diff
@@ -319,6 +356,68 @@ def update(self):
         _LOGGER.debug("Updated SolarEdge inventory: %s, %s", self.data, self.attributes)
 
 
+class SolarEdgeEnergyDetailsService(SolarEdgeDataService):
+    """Get and update the latest power flow data."""
+
+    def __init__(self, api, site_id):
+        """Initialize the power flow data service."""
+        super().__init__(api, site_id)
+
+        self.unit = None
+
+    @Throttle(ENERGY_DETAILS_DELAY)
+    def update(self):
+        """Update the data from the SolarEdge Monitoring API."""
+        try:
+            now = datetime.now()
+            today = date.today()
+            midnight = datetime.combine(today, datetime.min.time())
+            data = self.api.get_energy_details(
+                self.site_id,
+                midnight,
+                now.strftime("%Y-%m-%d %H:%M:%S"),
+                meters=None,
+                time_unit="DAY",
+            )
+            energy_details = data["energyDetails"]
+        except KeyError:
+            _LOGGER.error("Missing power flow data, skipping update")
+            return
+        except (ConnectTimeout, HTTPError):
+            _LOGGER.error("Could not retrieve data, skipping update")
+            return
+
+        if "meters" not in energy_details:
+            _LOGGER.debug(
+                "Missing meters in energy details data. Assuming site does not have any"
+            )
+            return
+
+        self.data = {}
+        self.attributes = {}
+        self.unit = energy_details["unit"]
+        meters = energy_details["meters"]
+
+        for entity in meters:
+            for key, data in entity.items():
+                if key == "type" and data in [
+                    "Production",
+                    "SelfConsumption",
+                    "FeedIn",
+                    "Purchased",
+                    "Consumption",
+                ]:
+                    energy_type = data
+                if key == "values":
+                    for row in data:
+                        self.data[energy_type] = row["value"]
```

Comment:
```suggestion
                        self.data[energy_type] = int(row["value"])
```
and this:
https://github.com/home-assistant/core/blob/1989c317679fff3fdc8f633f2c9b536ed4abd78e/homeassistant/components/solaredge/sensor.py#L275-L279
to:
``` python
        for key, value in overview.items():
            if key in ["lifeTimeData", "lastYearData", "lastMonthData", "lastDayData"]:
                data = int(value["energy"])
            elif key in ["currentPower"]:
                data = int(value["power"])
```